File: go/bob/bob.go
// Package bob Lackadaisical teenager stuff
package bob

import "strings"

// Hey Whatever man
func Hey(remark string) string {
	isQuestion := strings.HasSuffix(strings.TrimSpace(remark), "?")
	isShouting := strings.ToUpper(remark) == remark && strings.ToLower(remark) != remark
	isSilence := strings.TrimSpace(remark) == ""
	switch {
	case isQuestion && !isShouting:
		return "Sure."
	case isShouting && !isQuestion:
		return "Whoa, chill out!"
	case isQuestion && isShouting:
		return "Calm down, I know what I'm doing!"
	case isSilence:
		return "Fine. Be that way!"
	default:
		return "Whatever."
	}
}

File: python/bob/bob.py
def response(hey_bob):
    if hey_bob.isspace() or hey_bob == "":
        return "Fine. Be that way!"
    hey_bob = hey_bob.strip()
    isQuestion = hey_bob[-1] == "?"
    isShouting = hey_bob.isupper()
    if isQuestion and not isShouting:
        return "Sure."
    elif isQuestion and isShouting:
        return "Calm down, I know what I'm doing!"
    elif isShouting:
        return "Whoa, chill out!"
    return "Whatever."
